ppl/zqd/core.go

```diff
@@ -32,10 +32,14 @@ const indexPage = `
 </html>`
 
 type Config struct {
-	Logger      *zap.Logger
-	Personality string
-	Root        string
-	Version     string
+	Logger       *zap.Logger
+	Personality  string
+	Recruiter    string
+	Root         string
+	SpecPodIP    string
```

Comment:
The IP and Node are only used by workers, and I think the code would be easier to understand if their names reflect their usage, instead of their particular k8s sources.

How about creating WorkerConfig struct, with field names like "Address" and "Node", then making adding that into the overall Config struct?

Reply:
I will introduce a type WorkerConfig with {Host, Node}
We have used address for strings of the form host:port, but the SpecPodIP is just the host portion, in either IP format or as a dns address.
SpecNodeName will only be used an a logical identifier by us, even though it can have the form of a dns address, because it is not the in-cluster dns address.

Reply:
While we're here, is plumbing these through worth the effort?

When a worker registers with the recruiter, its IP address is already available to the recruiter via http.Request.RemoteAddr, so it doesn't actually need to be include in the request body. And if we want to include it there anyway, we can get it from httpd.Server.LocalAddr.

As for the node name, that can always be looked up using the worker's IP address.